plugin/src/main/java/org/opensearch/ml/action/agents/TransportRegisterAgentAction.java

```diff
@@ -60,10 +66,42 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLRegi
 
     private void registerAgent(MLAgent agent, ActionListener<MLRegisterAgentResponse> listener) {
         Instant now = Instant.now();
-        MLAgent mlAgent = agent.toBuilder().createdTime(now).lastUpdateTime(now).build();
+        String llmId;
+        if (agent.getType().equalsIgnoreCase(MLAgentType.CONVERSATIONAL.name()) && agent.getLlm() != null) {
+            llmId = agent.getLlm().getModelId();
+        } else {
+            llmId = null;
+        }
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            if (llmId != null) {
+                String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
+                mlModelManager.getModel(llmId, null, excludes, ActionListener.wrap(mlModel -> {
```

Comment:
Few tools also depends upon model ID. Validity of those model ID is not being checked. What is the plan to add tool validations?

Reply:
Are you saying there could be a use case, where we aren't using any hidden model id in agent, but using hidden model id in tools?

Reply:
yes, so it is better to hide the agent in general and not just hide the conversational model.

Reply:
Yeah, I'm going to hide hidden agent in general.

Reply:
addressed